cli/command/image/build/context.go

```diff
@@ -41,6 +41,20 @@ func ValidateContextDirectory(srcPath string, excludes []string) error {
 	if err != nil {
 		return err
 	}
+
+	pm, err := fileutils.NewPatternMatcher(excludes)
+	if err != nil {
+		return err
+	}
+	matches := func(file string) (bool, error) {
+		file = filepath.Clean(file)
+		if file == "." {
+			// Don't let them exclude everything, kind of silly.
```

Comment:
I'm not sure we can ignore this case; I know there's repositories using this (i.e. have `.dockerignore` with `.` in it, to not send anything)

Reply:
https://github.com/docker/cli/blob/master/vendor/github.com/docker/docker/pkg/fileutils/fileutils.go#L217-L232
it is original implementation.

Reply:
Hm... interesting. I definitely have seen people using this in their `.dockerignore`. Wondering if it worked at all for them (haven't tried it)

Reply:
I think we should add a test case here, or at least copy the tests from https://github.com/moby/moby/blob/master/pkg/fileutils/fileutils_test.go#L221 (as the code was copied too), and maybe it will need to extract this inlined function.

Reply:
i add test cases